pkg/kubecost/asset.go

```diff
@@ -2902,27 +2904,34 @@ type Diff[T any] struct {
 	Kind   DiffKind
 }
 
-// DiffAsset takes two AssetSets and returns a slice of Diffs by checking
-// the keys of each AssetSet. If a key is not found, a Diff is generated
-// and added to the slice.
-func DiffAsset(before, after *AssetSet) []Diff[Asset] {
-	changedItems := []Diff[Asset]{}
+// DiffAsset takes two AssetSets and returns a map of keys to Diffs by checking
+// the keys of each AssetSet. If a key is not found or is found with a different total cost,
+// a Diff is generated and added to the map.
+func DiffAsset(before, after *AssetSet, ratioCostChange float64) (map[string]Diff[Asset], error) {
```

Comment:
Let's add a bit to the comment about the cost change. Something like "If an asset exists in before and after and the total cost after is greater than ratio*before total cost, the asset will be added to the diff"